### ISSUE TITLE
weechat fails to build on OS X

### ISSUE THREAD

**Issue by sickill** [NixOS/nixpkgs#9156]
I tried installing WeeChat on OS X with `nix-env -i weechat` and I got this:

```
[ 67%] Building C object src/plugins/python/CMakeFiles/python.dir/weechat-python.o
[ 67%] Building C object src/plugins/python/CMakeFiles/python.dir/weechat-python-api.o
Linking C shared module python.so
Undefined symbols for architecture x86_64:
  "_PyUnicodeUCS4_AsUTF8String", referenced from:
      _weechat_python_unicode_to_string in weechat-python.o
      _weechat_python_dict_to_hashtable in weechat-python.o
      _weechat_python_exec in weechat-python.o
  "_PyUnicodeUCS4_FromString", referenced from:
      _weechat_python_init_module_weechat in weechat-python.o
ld: symbol(s) not found for architecture x86_64
clang-3.6: error: linker command failed with exit code 1 (use -v to see invocation)
src/plugins/python/CMakeFiles/python.dir/build.make:111: recipe for target 'src/plugins/python/python.so' failed
make[2]: *** [src/plugins/python/python.so] Error 1
CMakeFiles/Makefile2:895: recipe for target 'src/plugins/python/CMakeFiles/python.dir/all' failed
make[1]: *** [src/plugins/python/CMakeFiles/python.dir/all] Error 2
Makefile:136: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/z3lqpc773yab38bkcjb2kmm9s1paqld6-weechat-1.2.drv’ failed with exit code 2
error: build of ‘/nix/store/z3lqpc773yab38bkcjb2kmm9s1paqld6-weechat-1.2.drv’ failed
```

That's on OS X 10.10, nix 1.9

**Comment by flazz** [comment]
bump

**Comment by vcunat** [comment]
@flazz: it *is* in NixOS/nixpkgs ;-)

**Comment by boj** [comment]
Ran into this as well.

**Comment by henrytill** [comment]
This seems to be fixed in master.  Could you check  again with a current clone of nixpkgs and Nix HEAD?  

To install Nix HEAD:
```
git clone https://github.com/NixOS/nix
cd nix
nix-env -f release.nix -iA build.x86_64-darwin --option build-use-chroot false
```
Also, you should set
```
build-use-chroot = true
```
in `/etc/nix/nix.conf`

Instructions for working from a clone here: 
https://nixos.org/wiki/Nix_on_OS_X

**Comment by sickill** [comment]
It built successfully but... when I quit weechat I'm getting:

```
~ % weechat

*** Very bad! WeeChat is crashing (SIGSEGV received)
*** Full crash dump was saved to /Users/sickill/.weechat/weechat_crash_20151111_51815.log file.
***
*** Please help WeeChat developers to fix this bug:
***   1. If you have a core file, please run:  gdb /path/to/weechat core
***      then issue command: "bt full" and send result to developers
***      (see user's guide for more info about report of crashes).
***   2. Otherwise send backtrace (below), only if it is a complete trace.
***      Keep the crash log file, just in case developers ask you some info
***      (be careful, private info like passwords may be in this file).

======= WeeChat backtrace =======
(written by WeeChat 1.3, compiled on Jan 01 1970 00:00:01)
======= End of  backtrace =======
```

**Comment by Profpatsch** [comment]
(triage) does it still fail?

**Comment by andrewrmiller** [comment]
@sickill The issue you were seeing is only related to guile and weechat, not the nix build. You can find more information over at https://github.com/weechat/weechat/issues/769.

As for the actual nix build it is still failing for me with:

[ 64%] Building C object src/plugins/python/CMakeFiles/python.dir/weechat-python.o
[ 65%] Building C object src/plugins/python/CMakeFiles/python.dir/weechat-python-api.o
[ 65%] Linking C shared module python.so
ld: framework not found python
clang-3.7: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/plugins/python/CMakeFiles/python.dir/build.make:122: src/plugins/python/python.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:978: src/plugins/python/CMakeFiles/python.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
builder for ‘/nix/store/x12k6ksb1a75x17f7r8cw0af4jgr06hn-weechat-1.5.drv’ failed with exit code 2
error: build of ‘/nix/store/x12k6ksb1a75x17f7r8cw0af4jgr06hn-weechat-1.5.drv’ failed

This issue is resolved by the change proposed by @tfheen in #13032

I think that change should be integrated as it seems to solve the build problems on OS X, but I haven't tested this on NixOS or Linux.

**Comment by grahamc** [comment]
@tfheen's fix in #13032 compiles on linux too, but that cmake flag should probably be put behind a isDarwin flag anyway.

**Comment by copumpkin** [comment]
I think this should work now, so I'm closing. Please comment here if you still see the issue.
